Marlin/src/gcode/sd/M1001.cpp

```diff
@@ -80,6 +81,9 @@ void GcodeSuite::M1001() {
   // Stop the print job timer
   gcode.process_subcommands_now_P(PSTR("M77"));
 
+  // Stop all heaters (hot bed)
+  IF_DISABLED(SD_ABORT_NO_COOLDOWN, thermalManager.disable_all_heaters());
```

Comment:
I certainly considered all the points you have brought up before I posted the basic objection, and I stand by the user being allowed to leave the heaters on if that is their wish.

If some slicer profile is failing to turn off heaters at the end of the print job, that is a bug in the slicer profile. There is no guarantee that any 3D printer will turn off its heaters at the completion of a single SD file, and when batch printing it can speed things up to leave heaters on, or set them to some 'standby' temperature between batches.

As pointed out below, there are other safety features which can be used to prevent a neglected machine from leaving heaters on indefinitely. We could certainly add some `#warning` messages to the build to inform users that the machine firmware is –by design– dumb and obedient, so they should enable certain safety features to deal make an unattended machine more safe.

Reply:
You're right, that should be a good solution.